pytest/tests/mocknet/mirror.py

```diff
@@ -219,13 +220,55 @@ def reset_cmd(args, traffic_generator, nodes):
         )
         if sys.stdin.readline().strip() != 'yes':
             sys.exit()
+    if args.backup_id is None:
+        backups = nodes[0].neard_runner_ls_backups()
+        backups_msg = 'ID |  Time  | Description\n'
+        if 'start' not in backups:
+            backups_msg += 'start | None | initial test state after state root computation\n'
+        for backup_id, backup_data in backups.items():
+            backups_msg += f'{backup_id} | {backup_data.get("time")} | {backup_data.get("description")}\n'
+
+        print(f'Backups as reported by {nodes[0].name()}):\n\n{backups_msg}')
+        print('please enter a backup ID here:')
+        args.backup_id = sys.stdin.readline().strip()
+        if args.backup_id != 'start' and args.backup_id not in backups:
+            print(
+                f'Given backup ID ({args.backup_id}) was not in the list given')
+            sys.exit()
+
     all_nodes = nodes + [traffic_generator]
-    pmap(lambda node: node.neard_runner_reset(), all_nodes)
+    pmap(lambda node: node.neard_runner_reset(backup_id=args.backup_id),
+         all_nodes)
     logger.info(
         'Data dir reset in progress. Run the `status` command to see when this is finished. Until it is finished, neard runners may not respond to HTTP requests.'
     )
 
 
+def make_backup_cmd(args, traffic_generator, nodes):
+    if not args.yes:
+        print(
+            'this will stop all nodes and create a new backup of their home dirs. continue? [yes/no]'
+        )
+        if sys.stdin.readline().strip() != 'yes':
+            sys.exit()
+
+    if args.backup_id is None:
+        print('please enter a backup ID:')
+        args.backup_id = sys.stdin.readline().strip()
+        if re.match(r'^[0-9a-zA-Z.][0-9a-zA-Z_\-.]+$', args.backup_id) is None:
+            sys.exit('invalid backup ID')
+    if args.description is None:
+        print('please enter a description (enter nothing to skip):')
+        description = sys.stdin.readline().strip()
+        if len(description) > 0:
+            args.description = description
```

Comment:
I would only ask user for description if backup_id was also empty. This way it's possible to run this command without description in non-interactive mode.